.github/workflows/spelling3.yml

```diff
@@ -59,35 +83,41 @@ jobs:
       contents: read
       pull-requests: read
       actions: read
+      security-events: write
     outputs:
       followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-latest
-    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    if: ${{ contains(github.event_name, 'pull_request') || github.event_name == 'push' }}
     concurrency:
       group: spelling-${{ github.event.pull_request.number || github.ref }}
       # note: If you use only_check_changed_files, you do not want cancel-in-progress
       cancel-in-progress: true
     steps:
     - name: check-spelling
       id: spelling
-      uses: check-spelling/check-spelling@v0.0.21
+      uses: check-spelling/check-spelling@v0.0.22
       with:
-        suppress_push_for_open_pull_request: 1
+        suppress_push_for_open_pull_request: ${{ github.actor != 'dependabot[bot]' && 1 }}
         checkout: true
         check_file_names: 1
-        spell_check_this: check-spelling/spell-check-this@prerelease
+        spell_check_this: microsoft/winget-cli@master
         post_comment: 0
         use_magic_file: 1
-        extra_dictionary_limit: 10
+        warnings: bad-regex,binary-file,deprecated-feature,ignored-expect-variant,large-file,limited-references,no-newline-at-eof,noisy-file,non-alpha-in-dictionary,token-is-substring,unexpected-line-ending,whitespace-in-dictionary,minified-file,unsupported-configuration,no-files-to-check
+        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'microsoft' && 1 }}
+        use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
+        extra_dictionary_limit: 20
         extra_dictionaries:
-          cspell:cpp/src/compiler-msvc.txt
-          cspell:cpp/src/cpp.txt
-          cspell:cpp/src/stdlib-c.txt
+          cspell:software-terms/dict/softwareTerms.txt
           cspell:cpp/src/stdlib-cpp.txt
+          cspell:cpp/src/stdlib-c.txt
+          cspell:dotnet/dict/dotnet.txt
           cspell:filetypes/filetypes.txt
-          cspell:fullstack/fullstack.txt
-          cspell:powershell/powershell.txt
-          cspell:software-terms/src/software-terms.txt
+          cspell:csharp/csharp.txt
+          cspell:cpp/src/compiler-msvc.txt
```

Comment:
order doesn't matter, so I should probably shift things so that they aren't changing in this PR.